rest/codegen/paths.go

```diff
@@ -2,24 +2,57 @@ package codegen
 
 import (
 	"fmt"
-	"strings"
 	"text/template"
 
 	"goa.design/goa.v2/codegen"
 	goadesign "goa.design/goa.v2/design"
 	"goa.design/goa.v2/rest/design"
 )
 
+// note: conversion array to string path https://play.golang.org/p/0QHmyJeFhR
+
 const pathT = `{{range $i, $route := .Routes}}
 // {{$.EndpointName}}{{$.ServiceName}}Path{{if ne $i 0}}{{add $i 1}}{{end}} returns the URL path to the {{$.ServiceName}} service {{$.EndpointName}} HTTP endpoint.
-func {{$.EndpointName}}{{$.ServiceName}}Path{{if ne $i 0}}{{add $i 1}}{{end}}({{if .Arguments}}{{join .Arguments ", "}}{{end}}) string {
-{{- if .Params}}
-	return fmt.Sprintf("{{ .Path }}", {{join .Params ", "}})
+func {{$.EndpointName}}{{$.ServiceName}}Path{{if ne $i 0}}{{add $i 1}}{{end}}({{template "arguments" .Arguments}}) string {
+{{- if .Arguments}}
+	{{template "slice_conversion" .Arguments -}}
+	return fmt.Sprintf("{{ .Path }}"{{template "fmt_params" .Arguments}})
 {{- else}}
 	return "{{ .Path }}"
 {{- end}}
 }
-{{end}}
+{{end -}}
+
+{{- define "arguments" -}}
+{{range $i, $arg := . -}}
+{{if ne $i 0}}, {{end}}{{.Name}} {{goTypeRef .Type}}
+{{- end}}
+{{- end}}
+
```

Comment:
I was referring to the blank lines in between each block or are these also trimmed?